shared/libraries/websocket_streaming/src/websocket_client_device_impl.cpp

```diff
@@ -77,13 +77,18 @@ void WebsocketClientDeviceImpl::createWebsocketStreaming()
     websocketStreaming = WebsocketStreaming(streamingClient, connectionString, context);
 }
 
-void WebsocketClientDeviceImpl::onNewSignal(const StringPtr& signalId, const SubscribedSignalInfo& sInfo)
+void WebsocketClientDeviceImpl::onDataDescriptor(const StringPtr& signalId, const SubscribedSignalInfo& sInfo)
 {
     if (!sInfo.dataDescriptor.assigned())
         return;
 
     if (auto signalIt = deviceSignals.find(signalId); signalIt != deviceSignals.end())
     {
+        // sets signal name as it appeared in descriptor
+        auto protectedObject = signalIt->second.asPtr<IPropertyObjectProtected>();
+        if (sInfo.dataDescriptor.getName().assigned())
+            protectedObject.setProtectedPropertyValue("Name", sInfo.dataDescriptor.getName());
+
```

Comment:
`dataDescriptor.getName()` does not receive the name of the signal. The name of the signal is received within openDAQ as `signal.getName()`. At least the initial name should probably be transferred as metadata - and it's different from the signal ID.

Reply:
I aimed to address Nils' request "_It would also nice, if for a pure caps=WS device, we see also here the signal name not the id_" and make gui_demo app display signal name instead of weird signal id containing `#` symbol. gui demo displays the string obtained from `signal.getName()` which is the value of `"Name"` property.  In turn, this property is initialized with `localId` value.

Reply:
Sure, but I don't think the name on the data descriptor is correct here. Is only an ID available on the streaming client, and no signal "display name" ?

Reply:
signal name is available in `definition` json object on streaming client. For example it is `"name":"Connector1.Channel1.Signal2"` for fusion device and `"name":"DBX_CH1.value"` for mbedOS device. Signal generator tool keeps it default `"name":"value"`. openDAQ streaming server sets this field with name value from descriptor.

Reply:
Changed by [commit](https://github.com/openDAQ/openDAQ/pull/33/commits/685c9884dc1d3f799f6d784edcc4fedbd402859a). Now signal name is used as `"name"` field in streaming `"definition"`meta-data. The descriptor name is encoded into `"desc_name"` field of `"interpretation"` object for openDAQ->openDAQ communication.